### PR TITLE
NPCMoreTalkHandler: don't underflow selection

### PR DESCRIPTION
**Pull request by yuzumika**
Made this change to allow more than 127 hairs/faces to be available at the stylist NPCs, but it could be useful for other custom dialogues that involve many choices. Given that `selection` can also be read from the packet as an int, I think this change makes sense.

I'm not sure when exactly the client sends a byte vs an int for `selection`, so I've only tested it for my own use case. My guess is that a negative `selection` value is never desirable, so I think this change should be pretty safe.

### CODE REVIEW

**Comment by Arnuh** [comment]
This should just be a readUnsignedByte call

**Comment by yuzumika** [comment]
> This should just be a readUnsignedByte call

Thanks! I've implemented it that way instead.

File: src/main/java/net/packet/ByteBufInPacket.java [hunk 1]
```diff
@@ -22,6 +22,8 @@ public byte[] getBytes() {
     public byte readByte() {
         return byteBuf.readByte();
     }
+    @Override
+    public short readUnsignedByte() { return byteBuf.readUnsignedByte(); }
```
Comment (P0nk) [inline src/main/java/net/packet/ByteBufInPacket.java:26]: Please add a test, I'd like `ByteBufInPacket` to be fully covered by unit tests.

**Review by P0nk — CHANGES_REQUESTED** [review]

File: src/main/java/net/packet/ByteBufInPacket.java (hunk 1 above)
Comment (P0nk) [reply src/main/java/net/packet/ByteBufInPacket.java:26]: There are already a bunch of tests in `ByteBufInPacketTest.java`, so it should be easy to add.

Comment (yuzumika) [reply src/main/java/net/packet/ByteBufInPacket.java:26]: Done! Added 3 unit tests for readUnsignedByte, particularly around the boundaries.

**Review by P0nk — APPROVED** [review]

**Pull request merged**